source/lambda/kendraIndexPoller/lambda_function.py

```diff
@@ -0,0 +1,75 @@
+import boto3
+import json
+import datetime
+
+def convert_datetime_to_string(obj):
+    if isinstance(obj, datetime.datetime):
+        return obj.__str__()
+
+def on_create(event, context):
+    kendra_client = boto3.client('kendra')
+    # get status of index
+    response = kendra_client.describe_index(Id=event['PhysicalResourceId'])
+    is_complete = False
+    status = response['Status']
+    if status == "CREATING":
+        print("still creating kendra index")
+    elif status == "ACTIVE":
+        print("Kendra index is now active")
+        return { 'IsComplete': True }
+    elif status == "FAILED":
+        # throw an error
+        raise Exception("Kendra index creation failed with reason: {}".format(response['ErrorMessage']))
+    elif status == "DELETING" or status == "SYSTEM_UPDATING":
+        #print("Kendra index creation shows inconsistent status code, please fix and try again")
+        raise Exception("Kendra index creation shows inconsistent status code, please fix and try again. Reason:{}".format(response['ErrorMessage']))
+    return { 'IsComplete': is_complete }
+
+def on_delete(event, context):
+    kendra_client = boto3.client('kendra')
+    DUSkendraIndexId = event['PhysicalResourceId']
+    # check if the list_indices has the index id, if yes, then check status
+    kendra_indices = kendra_client.list_indices()
+    kendra_indices = json.loads(json.dumps(kendra_indices,default = convert_datetime_to_string))
+    kendra_index_ids = []
+    for index in kendra_indices['IndexConfigurationSummaryItems']:
+        kendra_index_ids.append(index['Id'])
+    # if the index id is not present, it has been deleted
+    if DUSkendraIndexId not in kendra_index_ids:
+        print("Kendra index with id {} deleted".format(DUSkendraIndexId))
+        return {'IsComplete': True}
+    if len(kendra_index_ids) > 0:
```

Comment:
Do you need this if condition?
We could just have the loop since it would end up doing the same functionality

Reply:
My mistake. I thought iterating over an empty list would give an error. But it does not. I will remove it.